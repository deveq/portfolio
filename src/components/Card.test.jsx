import { render, screen, fireEvent } from "@testing-library/react";

import Card from "./Card";
import "./card.css";

describe("<Card /> component", () => {
  const text = "ChildComponent";

  const ChildComponent = () => {
    return <div>{text}</div>;
  };

  it("기본 렌더링 테스트", () => {
    const className = "test-className";
    const { container } = render(
      <Card className={className}>
        <ChildComponent />
      </Card>,
    );

    const childComponent = screen.getByText(text);
    expect(childComponent.parentElement.classList.contains(className)).toBe(
      true,
    );
    expect(childComponent).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("클릭 이벤트 테스트", () => {
    const handleClick = jest.fn();

    render(
      <Card onClick={handleClick}>
        <ChildComponent />
      </Card>,
    );

    const childComponent = screen.getByText(text);
    expect(handleClick).toHaveBeenCalledTimes(0);

    fireEvent.click(childComponent);
    expect(handleClick).toHaveBeenCalledTimes(1);
    fireEvent.click(childComponent);
    expect(handleClick).toHaveBeenCalledTimes(2);
  });
});
