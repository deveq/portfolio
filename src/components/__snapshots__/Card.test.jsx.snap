// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> component 기본 렌더링 테스트 1`] = `
<div>
  <article
    class="card test-className"
  >
    <div>
      ChildComponent
    </div>
  </article>
</div>
`;
